app/graphql/resolvers/article_query_function.rb

```diff
@@ -0,0 +1,12 @@
+class Resolvers::ArticleQueryFunction < GraphQL::Function
+
+    def select_published(articles)
+        # nil articles will be filtered out
+        if articles.nil?
+            return nil
+        end
+        
+        return articles.where(is_published: true)

```

Comment:
imo this should be in a model

Reply:
Could you explain exactly what you mean? I'm not familiar with models. Thanks!

Reply:
In Rails there's this idea that you should have thin controllers and fat models. Models contain the logic for the actual piece of data, while controllers contain the logic for connecting the models to the views. In general, controllers/resolvers shouldn't have a whole lot of logic in them. Instead, it should be encapsulated in the model. The model is in `app/models/article.rb`. In this case, if you want to get the published articles, maybe you want to make a class method like:

```
class Article

def published()
  Article.where(is_published: true)
end

end
```
Then you can do:
```
Article.published.first(10)
```

Or whatever

Reply:
That's a lot nicer, I changed all my code to use that, and added an unpublished method as well.